enhancements/scheduling/descheduler-profiles.md

```diff
@@ -0,0 +1,259 @@
+strategies into existing groups as well.
+
+Below are the proposed initial profiles for the currently available descheduling strategies:
+
+* `AffinityAndTaints`: enables `RemovePodsViolatingInterPodAntiAffinity`, `RemovePodsViolatingNodeAffinity`,
+and `RemovePodsViolatingNodeTaints`. These are the most basic descheduling strategies and most likely the minimum for
+what every user of the Descheduler will want to run. In the future, this could be split into 2 profiles (for hard vs. soft
+affinity requirements).
+
+* `TopologyAndDuplicates`: enables `RemovePodsViolatingTopologySpreadConstraint` and `RemoveDuplicates`.
+These strategies are focused specifically on spreading pods evenly among nodes.
+
+* `LifecycleAndUtilization`: enables `RemovePodsHavingTooManyRestarts`, `LowNodeUtilization`,
+and `PodLifeTime`. These focus on the lifecycle of pods and nodes.
+
+These profiles each serve distinct, unrelated functions so users will not be limited to enabling
+just one. There is no risk of interference between the profiles so any combination of them can
+be enabled at once.
+
+### User Stories [optional]
+
+#### Story 1
+
+As a sysadmin, I want to ensure that my running pods respect node taints, affinity, and inter-pod
+affinity. I enable the `AffinityAndTaints` profile to ensure this.
+
+#### Story 2
+
+As a sysadmin, I have a low risk of affinity and taints changing after my pods are scheduled
+but I do want to ensure that they are evenly-distributed among the nodes of the cluster. I also
+want to keep node utilization balanced. So I enable both `TopologyAndDuplicates` and `LifecycleAndUtilization`.
+
+### Risks and Mitigations
+
+* this will restrict the configuration options from what is currently available in the descheduler operator, but
+since it is currently in tech preview (and the API is only beta) this should not be an issue
+* this will improve stability and security by restricting config to only what we are prepared to support
+
+## Design Details
+
+The new field will be added to the existing operator spec:
+```go
+// KubeDeschedulerSpec defines the desired state of KubeDescheduler
+type KubeDeschedulerSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+  ...
+  Profiles DeschedulerProfiles `json:"profiles"`
+  ...
+}
+
+// DeschedulerProfiles defines the list of available profiles and allows enabling/disabling them
+type DeschedulerProfiles struct {
+     AffinityAndTaints       bool `json:"affinityAndTaints"`
+     TopologyAndDuplicates   bool `json:"topologyAndDuplicates"`
+     LifecycleAndUtilization bool `json:"lifecycleAndUtilization"`
+}
+```
+
+This approach will clearly present users with all of their options for configuration
+while eliminating the chance of typos and the need for validation against input.
+
+An alternative would be to implement this strictly as a list:
+```go
+type KubeDeschedulerSpec struct {
+  Profiles []string `json:"profiles"`
+  ...
+}
```

Comment:
Another alternative is a list of enums.

Reply:
are there any opinions on preference for either one? I'm not attached either way but curious if anyone else had any insight

Reply:
let's go with a list of enums, we can then re-use the kubebuilder validation for enums for that type. The only thing we'll have to ensure is to de-duplication or call out what happens and validation if contradicting profiles are applied.